client/script.js

```diff
@@ -0,0 +1,114 @@
+function populateTodoList(todos) {
+  try {
+    let list = document.getElementById("todoList");
+
+    // Clear the contents of the list before creating new items
+    list.innerHTML = "";

```

Comment:
making innerHTML = "" is a bit "hacky" way and not the best approach.
in fact, you won't be able to do it in React. It doesn't allow it.
better way is using `replaceChildren()` methid - you can pass an empty array into that method:  
```js
list.replaceChildren([]);
```